src/sample.cpp

```diff
@@ -93,20 +97,20 @@ SampleResult RulesSampler::sample(const std::string& environment, const std::str
   // `SamplingPriority::SamplerDrop`.
 
   SampleResult result;
-  result.rule_rate = rule_result.rate;
+  result.applied_rate = result.rule_rate = rule_result.rate;
+  result.sampling_mechanism = SamplingMechanism::Rule;
   auto max_hash = maxIdFromSampleRate(rule_result.rate);
   uint64_t hashed_id = trace_id * constant_rate_hash_factor;
   if (hashed_id >= max_hash) {
     result.sampling_priority = std::make_unique<SamplingPriority>(SamplingPriority::UserDrop);
     return result;
   }
 
-  // Even though both the priority sampler and the matching sampling rule,
-  // above, did not drop this span, we still might drop the span in order to
-  // satify the configured maximum sampling rate for spans selected by rule
-  // based sampling overall.
+  // Even though the matching sampling rule, above, did not drop this span, we
+  // still might drop the span in order to satify the configured maximum
+  // sampling rate for spans selected by rule based sampling overall.
   auto limit_result = sampling_limiter_.allow();
-  result.limiter_rate = limit_result.effective_rate;
+  result.applied_rate = result.limiter_rate = limit_result.effective_rate;
```

Comment:
Caleb and I had a discussion about whether to use the configured rule sampler rate here (or perhaps, at least only for purposes of `UpstreamService`), or whether to use the limiter's effective rate.

Probably doesn't matter, but man so complicated.